### DISCUSSION TITLE
Storage mode for boats and RVs

### DISCUSSION THREAD

**Discussion by Mike101202** [Louisvdw/dbus-serialbattery#1064] [Ideas]
I’ve been using your BMS driver on my boat for almost 3 years. It has been simply awesome.

Ok, many lifepo4 battery systems are being installed on boats. One of the main issues with putting a lithium system on a boat is the sporadic nature of boat usage. The boat might be used on the weekend, or the boat may be put in storage for a month or longer. With lead batteries, this was easy. Just put a trickle charger on it, and keep it at 100%. Easy. 
With lithium, that will kill the batteries in a very short amount of time. 

Lithium likes  being stored at about 60 to 70%, depending on the manufacturer and the literature you read. But definitely not 100%

So as a workaround, I have 2 config.ini files. 
One labeled configStorage.ini and the other configNormal.ini each file has a different charge curve, so I can have the batteries charging to 100% for normal use, and only charge to 60% when I leave the boat in a marina on shore power. 

Yep this is really clunky. Each time I have to SSH and rename the appropriate file from configStorage.ini to config.ini so the batteries operate to full charge when we are on the boat.

ok, my suggestion is obvious.. is it possible to select from 2 different charges profiles from the cerbo panel? 

My feeling is that there are a lot of lithium batteries that are being destroyed by being kept at 100% . This change would save many lithium batteries from an early demise and greatly improve the options of your great program. 

To be honest, I’m very surprised that it not a standard function of victron products. 

Thanks..

Mike Rose

**Comment by mr-manuel** [comment]
Why have you this complex configuration? How does your config look like? In default settings the battery switches to float and stays in float which can also seen as storage mode.

**Reply by Mike101202** [reply]
The issue is if I drop the float voltage down, the batteries will still be fully charged bu the time float kicks in. 
This means they will be sitting at 100% for long periods, which is not good for them. This is why I have 2 charge profiles. the storage profile stops charging them when they get to 3.37v

Here is my config ini file. as you can see I have a storage ini and a normal ini. they are the same, I just don't have to hash out the values i don't want.

What are your thoughts?

**Comment by Mike101202** [comment]
![image](https://github.com/Louisvdw/dbus-serialbattery/assets/167390447/6eab5511-6821-4423-aa71-338ef04c3484)
![image](https://github.com/Louisvdw/dbus-serialbattery/assets/167390447/74973592-2508-412e-ad08-2f085fe83e73)

**Reply by Mike101202** [reply]
The system has 5 * 200AH 4 cell batteries

**Comment by Mike101202** [comment]
![image](https://github.com/Louisvdw/dbus-serialbattery/assets/167390447/67cb0fd3-e98f-4c65-9d00-1d8803ef3c24)
![image](https://github.com/Louisvdw/dbus-serialbattery/assets/167390447/078c54c8-a7af-4b9b-870d-ce82982ee9d3)

**Comment by mr-manuel** [comment]
If there are enough votes for this I will implement it. As it is now you can use Node-RED to achieve your request and without the need to connect through SSH.

**Reply by Mike101202** [reply]
Thanks!!

**Reply by ofernander** [reply]
> If there are enough votes for this I will implement it. As it is now you can use Node-RED to achieve your request and without the need to connect through SSH.

If it's easy to implement it would be a nice feature. I'm also using this driver on a boat and do almost the same as @Mike101202. Using node-red is an option but I would rather keep things "in house" software wise as it reduces the failure points. 

@mr-manuel and thanks for all the awesome work on this! Probably the best piece of software in this category around.

**Comment by mr-manuel** [comment]
## Crowdfunding a feature request

I was thinking of a possibility to allow not most wanted features to be developed even, if they don't get enough votes. So the idea is to have something like a crowd funding for each feature. This allows to trigger it's development by votes or reaching the fund goal. Unfortunately I found no plattform which gives this possibility like there was once, so we make that via PayPal donations. After a user donates for a feature I will add manually a comment, so it's current state is known. If the user likes, he can also comment with the amount he donated.

Based on an effort estimate (without testing) this feature would take at least 2 hours to complete. With testing 4-6 hours. Based on a normal programmers hourly rate, with 2 hours that would be 200 €. Let's start with that to see, if and how this would work.

## How can you contribute to get this completed?
Make a donation via https://www.paypal.com/donate/?hosted_button_id=3NEVZBDM5KABW, select `dbus-serialbattery` and add `storage mode` as note. If you forgot to select `dbus-serialbattery` or to add the note reply to this answer with the amount you donated.

## Goal
0 € of 200 € reached.
